Automattic/woocommerce-payments
# Adds new UPE feature flag

Fixes #5682 

#### Changes proposed in this Pull Request


Adds a new feature flag for the new UPE/deferred intent feature to hide behind. 

Please see, review, and approve Automattic/woocommerce-payments-dev-tools#86 to assist with toggling this feature flag. This flag doesn't do anything yet, but will be used in #5770. Decisions on how enablement and disablement of this feature flag can be left for other issues.




#### Testing instructions


This change does not yet effect any changes, so you will have to check your `wp_options` table to ensure the flag has been added and enabled on your site.

1. Pull the changes in this PR and Automattic/woocommerce-payments-dev-tools#86 and ensure both are in use on your site.
2. Go to the Dev Tools, ensure the checkbox titled "Enable Split UPE checkout with deferred intent creation" is checked, then submit the page.
3. On refresh, this checkbox should remain checked.
4. In your site's `wp_options` search for the entry with `option_name` of ~`woocommerce_woocommerce_payments_settings`. Within this PHP serialised array, you should find a key with value of `_wcpay_feature_upe_deferred_intent` now~ `_wcpay_feature_upe_deferred_intent` with a value of `1`.
5. Return to dev tools, uncheck the same checkbox, and submit the page.
6. On refresh, the checkbox should remain unchecked. In your site's options table, ~in `woocommerce_woocommerce_payments_settings` object~, `_wcpay_feature_upe_deferred_intent` should now ~have a value of `0`~ be removed from the table.
7. Lucky number seven.


-------------------


**Post merge**
